src/hypofuzz/provider.py

```diff
@@ -133,6 +133,28 @@ class State:
 QueueElement = tuple[QueuePriority, ChoicesT, Optional[Any]]
 
 
+def bucket_target_value(v: Any) -> Any:
+    # technically, someone could do event("target", not_an_int) and get us
+    # confused with a target() call. We could/should track this provenance more
+    # carefully in observation.features
+    if not isinstance(v, (int, float)):
+        return v
+
+    # bucket by (base 2) orders of magnitude.
+    return int(math.log2(v))
```

Comment:
`OverflowError` for `v=1e999`, among other values.  

I think we'll need a more sophisticated approach here, possibly with dynamic bucketing... but for now I'd consider simply not treating targets as virtual branches, except for the current-largest value (breaking ties by choice-sequence sort order).